src/Gateway.php

```diff
@@ -0,0 +1,490 @@
+	 */
+	const SDK_VERSION = '5.14.0';
+
+	/**
+	 * Config.
+	 *
+	 * @var Config
+	 */
+	private $config;
+
+	/**
+	 * Client.
+	 *
+	 * @var Client
+	 */
+	public $client;
+
+	/**
+	 * Constructs and initializes an Adyen gateway.
+	 *
+	 * @param Config $config Config.
+	 */
+	public function __construct( Config $config ) {
+		$this->config = $config;
+
+		$this->set_method( self::METHOD_HTTP_REDIRECT );
+
+		$this->supports = [
+			'webhook_log',
+			'webhook',
+		];
+
+		$this->client = new Client( $config );
+	}
+
+	/**
+	 * Get supported payment methods
+	 *
+	 * @return array<string>
+	 * @see Core_Gateway::get_supported_payment_methods()
+	 */
+	public function get_supported_payment_methods() {
+		return [
+			PaymentMethods::AFTERPAY_COM,
+			PaymentMethods::ALIPAY,
+			PaymentMethods::APPLE_PAY,
+			PaymentMethods::BANCONTACT,
+			PaymentMethods::BLIK,
+			PaymentMethods::CREDIT_CARD,
+			PaymentMethods::DIRECT_DEBIT,
+			PaymentMethods::EPS,
+			PaymentMethods::GIROPAY,
+			PaymentMethods::GOOGLE_PAY,
+			PaymentMethods::IDEAL,
+			PaymentMethods::KLARNA_PAY_LATER,
+			PaymentMethods::KLARNA_PAY_NOW,
+			PaymentMethods::KLARNA_PAY_OVER_TIME,
+			PaymentMethods::MB_WAY,
+			PaymentMethods::SOFORT,
+			PaymentMethods::SWISH,
+			PaymentMethods::TWINT,
+			PaymentMethods::VIPPS,
+		];
+	}
+
+	/**
+	 * Get available payment methods.
+	 *
+	 * @return array<int, string>
+	 * @see Core_Gateway::get_available_payment_methods()
+	 */
+	public function get_available_payment_methods() {
+		$core_payment_methods = [];
+
+		$payment_methods_response = $this->client->get_payment_methods( new PaymentMethodsRequest( $this->config->get_merchant_account() ) );
+
+		foreach ( $payment_methods_response->get_payment_methods() as $payment_method ) {
+			$type = $payment_method->get_type();
+
+			if ( null === $type ) {
+				continue;
+			}
+
+			$core_payment_method = PaymentMethodType::to_wp( $type );
+
+			$core_payment_methods[] = $core_payment_method;
+		}
+
+		$core_payment_methods = array_filter( $core_payment_methods );
+		$core_payment_methods = array_unique( $core_payment_methods );
+
+		return $core_payment_methods;
+	}
+
+	/**
+	 * Get issuers.
+	 *
+	 * @return array<string, string>|array<int, array<string, array<string, string>>>
+	 * @see Core_Gateway::get_issuers()
+	 */
+	public function get_issuers() {
+		$issuers = [];
+
+		$payment_methods_response = $this->client->get_payment_methods( new PaymentMethodsRequest( $this->config->get_merchant_account() ) );
+
+		$payment_methods = $payment_methods_response->get_payment_methods();
+
+		// Limit to iDEAL payment methods.
+		$payment_methods = array_filter(
+			$payment_methods,
+			/**
+			 * Check if payment method is iDEAL.
+			 *
+			 * @param PaymentMethod $payment_method Payment method.
+			 * @return boolean True if payment method is iDEAL, false otherwise.
+			 */
+			function( $payment_method ) {
+				return ( PaymentMethodType::IDEAL === $payment_method->get_type() );
+			}
+		);
```

Comment:
We can use `PaymentMethodsRequest::set_allowed_payment_methods( [ PaymentMethodType::IDEAL ] )` to only request iDEAL issuers.

```suggestion
		$payment_methods_request = new PaymentMethodsRequest( $this->config->get_merchant_account() );

		$payment_methods_request->set_allowed_payment_methods( [ PaymentMethodType::IDEAL ] );
		
		$payment_methods_response = $this->client->get_payment_methods( $payment_methods_request );

		$payment_methods = $payment_methods_response->get_payment_methods();
```